### PR TITLE
Disk usage monitoring

### PR DESCRIPTION
**Pull request by mocsharp**
### Description
Introduce 2 new configurations, `watermarkPercent` and `reservedSpaceGB`:
* `watermarkPercent`: stops storing, exporting and retrieving when the used stroage space is above the watermark value.
* `reservedSpaceGB`: stops storing, exporting and retrieving when available storage space is less than the reserved space value.

User shall configure `reservedSpaceGB` based on size of a study, recommended value is 2-3 * the average study size * number of Clara AETs

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally by running `./src/run-tests-in-docker.sh`.
- [x] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [x] User guide updated.
- [ ] NGC publishing metadata updated.
- [x] I have updated the changelog

### CODE REVIEW

File: src/Server/Services/Disk/StoageInfoProvider.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        bool HasSpaceAvailableToStore { get; }
+        bool HasSpaceAvailableForExport { get; }
+        bool HasSpaceAvailableToRetrieve { get; }
+        long AvailableFreeSpace { get; }
+    }
+
+    public class StorageInfoProvider : IStorageInfoProvider
+    {
+        private const long OneGB = 1000000000;
```
Comment (whoisj) [inline src/Server/Services/Disk/StoageInfoProvider.cs:36]: a gigabyte is 1024 * 1024 * 1024, or 1,073,741,824 bytes; not 1000000000 bytes.

Comment (mocsharp) [reply src/Server/Services/Disk/StoageInfoProvider.cs:36]: `GB` (base 10) is 1000^3 and `GiB` (base 2) is 1024^3.

**Review by aihsani — CHANGES_REQUESTED** [review]

File: src/Server/Services/Disk/StoageInfoProvider.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mocsharp) [reply src/Server/Services/Disk/StoageInfoProvider.cs:36]: No :p

**Review by aihsani — CHANGES_REQUESTED** [review]

File: src/Server/Services/Http/HealthController.cs [hunk 2]
```diff
@@ -97,13 +104,15 @@ public ActionResult Ready()
...
-            services.Add(typeof(Jobs.DataRetrievalService));
-            services.Add(typeof(ScpService));
-            services.Add(typeof(Scu.ScuExportService));
-            services.Add(typeof(Export.DicomWebExportService));
+            var services = new List<Type>
+            {
+                typeof(Disk.SpaceReclaimerService),
+                typeof(Jobs.JobSubmissionService),
+                typeof(Jobs.DataRetrievalService),
+                typeof(ScpService),
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply src/Server/Services/Http/HealthController.cs]: Updated

File: docs/compliance/dicom.md [hunk 3]
```diff
@@ -24,7 +24,8 @@ SOP Classes.
 ### Association Policies
 
 * Clara DICOM Storage SCP accepts associations but does not initiate associations.
-* Clara DICOM Storage SCP accepts a maximum of 100 (configurable) concurrent associations.
+* Clara DICOM Storage SCP accepts 1000 (configurable) simultaneous associations across all configured AE Titles with default of 25.
+* Clara DICOM Storage SCP accepts associations when storage space usage is less than the configured watermark and the available storage space is above the configured reserved storage size.
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply docs/compliance/dicom.md]: Updated

File: docs/compliance/dicom.md [hunk 4]
```diff
@@ -24,7 +24,8 @@ SOP Classes.
 ### Association Policies
 
 * Clara DICOM Storage SCP accepts associations but does not initiate associations.
-* Clara DICOM Storage SCP accepts a maximum of 100 (configurable) concurrent associations.
+* Clara DICOM Storage SCP accepts 1000 (configurable) simultaneous associations across all configured AE Titles with default of 25.
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply docs/compliance/dicom.md]: Updated

**Review by MMelQin — APPROVED** [review]

**Review by aihsani — APPROVED** [review]

**Pull request merged**